experimental/packages/opentelemetry-instrumentation-http/src/http.ts

```diff
@@ -116,7 +115,9 @@ export class HttpInstrumentation extends InstrumentationBase<Http> {
       'http',
       ['*'],
       moduleExports => {
-        this._diag.debug(`Applying patch for http@${this._version}`);
+        this._diag.debug(
```

Comment:
I think we should use the version given in the callback above as we generally log the target package version (ex with grpc: https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-instrumentation-grpc/src/grpc/index.ts#L80)

Reply:
Ah right, I misread the purpose of this version here and thought it was supposed to be the instrumentation version, not the node:http version. I will update in a new commit 👍

Reply:
This has now been updated to reliably grab the Node version